qiskit/algorithms/gradients/base_qgt.py

```diff
@@ -0,0 +1,342 @@
+
+            rows, cols = np.triu_indices(len(parameter_indices))
+            for row, col in zip(rows, cols):
+                for g_parameter1, coeff1 in gradient_circuit.parameter_map[parameter_indices[row]]:
+                    for g_parameter2, coeff2 in gradient_circuit.parameter_map[parameter_indices[col]]:
+                        if isinstance(coeff1, ParameterExpression):
+                            local_map = {
+                                p: parameter_values_[circuit.parameters.data.index(p)]
+                                for p in coeff1.parameters
+                            }
+                            bound_coeff1 = coeff1.bind(local_map)
+                        else:
+                            bound_coeff1 = coeff1
+                        if isinstance(coeff2, ParameterExpression):
+                            local_map = {
+                                p: parameter_values_[circuit.parameters.data.index(p)]
+                                for p in coeff2.parameters
+                            }
+                            bound_coeff2 = coeff2.bind(local_map)
+                        else:
+                            bound_coeff2 = coeff2
+                        qgt[row, col] += (
+                            float(bound_coeff1)
+                            * float(bound_coeff2)
+                            * results.qgts[idx][g_parameter_indices[g_parameter1], g_parameter_indices[g_parameter2]]
+                        )
+            qgt += np.triu(qgt, k=1).conjugate().T
+            qgts.append(qgt)
+            metadata.append([{"parameters": parameter_indices}])
+        return QGTResult(
+            qgts=qgts, metadata=metadata, options=results.options
+        )
+
+    def _validate_arguments(
+        self,
+        circuits: Sequence[QuantumCircuit],
+        parameter_values: Sequence[Sequence[float]],
+        parameter_sets: Sequence[set[Parameter]],
+    ) -> None:
+        """Validate the arguments of the ``run`` method.
+
+        Args:
+            circuits: The list of quantum circuits to compute the QFIs.
+            parameter_values: The list of parameter values to be bound to the circuit.
+            parameters: The Sequence of Sequence of Parameters to calculate only the QFIs of
+                the specified parameters. Each Sequence of Parameters corresponds to a circuit in
+                ``circuits``. Defaults to None, which means that the QFIs of all parameters in
+                each circuit are calculated.
+
+        Raises:
+            ValueError: Invalid arguments are given.
+        """
+        if len(circuits) != len(parameter_values):
+            raise ValueError(
+                f"The number of circuits ({len(circuits)}) does not match "
+                f"the number of parameter value sets ({len(parameter_values)})."
+            )
+
+        if len(circuits) != len(parameter_sets):
```

Comment:
I am a bit confused by the docstring above. If the `parameter_sets` default to `None`, then this check should take that into account. Else, you can ignore my suggestion for the default, but the info should be removed from the docstring.